### PR TITLE
Move linting steps to GitHub Actions

### PR DESCRIPTION
**Pull request by tsatam**
### Which issue this PR addresses:

Fixes no Jira

### What this PR does / why we need it:

- [X] Moves Python linting/testing steps from ADO CI pipeline into a new `ci-python` GitHub Actions workflow
- [X] Removes Go unit testing/build steps in GitHub Actions ci-go workflow, as it is already performed in the ADO CI pipeline
- [X] Consolidates go linting steps into the `ci-go` GitHub Actions workflow

### Test plan for issue:

- [x] All GitHub Actions workflows and ADO pipelines pass (checks on this PR are green)

### Is there any documentation that needs to be updated for this PR?

No

### How do you know this will function as expected in production? 

Non-production change

### CODE REVIEW

**Review by SudoBrendan — DISMISSED** [review]
LGTM - this de-dpulicates/consolidates the PR validation pipeline and actions performed by `make ci-rp`. Code norms go in GitHub (e.g. linting), and production readiness goes in ADO (`make ci-rp`).

**Review by shubhadapaithankar — DISMISSED** [review]
LGTM

File: .pipelines/ci.yml [hunk 1]
```diff
@@ -26,78 +26,57 @@ resources:
...
+          repository: 'azext-aro'
+          pushLatest: true
 
-          # Push the image to ACR
-          - template: ./templates/template-acr-push.yml
-            parameters:
-              acrFQDN: 'arosvcdev.azurecr.io'
-              repository: 'azext-aro'
-              pushLatest: true
+  - job: Build_Lint_Test_RP_And_Portal
```
Comment (kimorris27) [inline .pipelines/ci.yml:49]: Shouldn't this ADO pipeline step be renamed?

```suggestion
  - job: Build_And_Test_RP_And_Portal
```

**Review by rhamitarora — APPROVED** [review]

File: .pipelines/ci.yml (hunk 1 above)
Comment (tsatam) [reply .pipelines/ci.yml:49]: Good suggestion

**Review by kimorris27 — APPROVED** [review]

**Review by shubhadapaithankar — APPROVED** [review]

**Review by kevinobriendotca — APPROVED** [review]

**Pull request merged**